packages/assets-controllers/src/multicall.ts

```diff
@@ -392,7 +392,27 @@ export const multicallOrFallback = async (
   }
 
   const multicallAddress = MULTICALL_CONTRACT_BY_CHAINID[chainId];
-  return await (multicallAddress
-    ? multicall(calls, multicallAddress, provider, maxCallsPerMulticall)
-    : fallback(calls, maxCallsParallel));
+  if (multicallAddress) {
+    try {
+      return await multicall(
+        calls,
+        multicallAddress,
+        provider,
+        maxCallsPerMulticall,
+      );
+    } catch (error: unknown) {
+      // Fallback only on revert
+      // https://docs.ethers.org/v5/troubleshooting/errors/#help-CALL_EXCEPTION
+      if (
+        !error ||
+        typeof error !== 'object' ||
+        !('code' in error) ||
+        error.code !== 'CALL_EXCEPTION'
```

Comment:
This seems like a complicated condition. Could you elaborate a bit on your code comment on line 404 to better explain what these conditions represent?

Reply:
Trying to only retry if the call reverted, and not retry on other kinds of errors that could occur.  I don't know exactly what I expect those errors to be, but could be like a dead rpc connection or something.